app/code/Magento/SalesRule/Model/ResourceModel/Rule/Collection.php

```diff
@@ -332,13 +332,9 @@ public function addWebsiteGroupDateFilter($websiteId, $customerGroupId, $now = n
                 'customer_group_ids.' .
                 $entityInfo['entity_id_field'] .
                 ' = cgw.' .
-                $entityInfo['entity_id_field'] . ' AND ' . $websiteId . ' = cgw.website_id',
+                $entityInfo['entity_id_field'] . ' AND ' . (int) $websiteId . ' = cgw.website_id',
```

Comment:
In my opinion there is not solving core of issue because it's looks like unpredictable application behavior.
Everywhere we expect to receive integer value of website id and we need to solve core of this problem specifically why website id is null.

@Sekiphp could you please add total list of steps for reproducing the error

Reply:
@Den4ik It is truth. I was not able to simulate this error, but this part is problematic and I just use same solution, which is used in method, which is also called inside this  method. In our website there are a lot of diferrent salesrules and probably not all are configured well and throws this error in some combination of steps.

Reply:
Hi @Sekiphp,
I just analyzed where this code is used - looks like it might be in https://github.com/magento/magento2/blob/3218b12a60e6174dfe02aef28ba6532dc63988dc/app/code/Magento/SalesRule/Block/Rss/Discounts.php#L97
While website id is getting retrieved from store. In case when store doesn't have linked website - this will trigger such error.
https://github.com/magento/magento2/blob/3218b12a60e6174dfe02aef28ba6532dc63988dc/app/code/Magento/SalesRule/Block/Rss/Discounts.php#L73
In this case, I think direct type conversion is good